Spatial.gd

```diff
@@ -818,27 +901,33 @@ func _on_TexturePathOpen_file_selected(path):
 # a path node is created.
 ################################################################################
 func _on_NewPath_pressed():
-	self.currently_active_path = null
+	self.currently_active_path_index = null
 
 ################################################################################
 # Create a new icon and give it the texture
 ################################################################################
 func _on_NewIcon_pressed():
 	var waypoint_category: Waypoint.Category = self.currently_active_category.get_metadata(0).waypoint_category
-	var waypoint_icon = waypoint_category.new_icon()
+	var waypoint_icon = waypoint_category.add_icon()
+	var position = waypoint_icon.new_position()
+	position.set_x(self.player_position.x)
+	position.set_y(self.player_position.y)
+	position.set_z(-self.player_position.z)
 	gen_new_icon(self.player_position, self.next_texture_path, waypoint_icon, self.currently_active_category)
 
 # A new path point is created
 func _on_NewPathPoint_pressed():
-	if self.currently_active_path == null:
+	if self.currently_active_path_index == null:
 		var waypoint_category: Waypoint.Category = self.currently_active_category.get_metadata(0).waypoint_category
-		var waypoint_trail = waypoint_category.new_trail()
+		var waypoint_trail = waypoint_category.add_trail()
+		var trail_data = waypoint_trail.new_trail_data()
+		trail_data.add_points_x(self.player_position.x)
+		trail_data.add_points_y(self.player_position.y)
+		trail_data.add_points_z(-self.player_position.z)
 		gen_new_path([self.player_position], self.next_texture_path, waypoint_trail, self.currently_active_category)
+		self.currently_active_path_index = len(waypoint_category.get_trail()) - 1
```

Comment:
This method of getting the index of the newly generated path seems a little hacky. If you keep this style of index then gen_new_path should probably return info about the new path that is used instead of relying on the new path to always be the last element of the waypont data.